src/context/values.js

```diff
@@ -174,12 +195,12 @@ export class Values {
   /**
    * Whether this node has inputs for the specified property.
    *
-   * @param {!ContextProp} prop
+   * @param {!ContextPropDef} prop
    * @return {boolean}
    */
   has(prop) {
-    const inputsByKey = this.inputsByKey_;
-    return !!inputsByKey && inputsByKey.has(prop.key);
+    const {key} = prop;
+    return !!this.inputsByKey_?.has(key);
```

Comment:
ignorable super nit: very slightly nicer as `prop.key`

Reply:
I agree! I think I assumed it was destructured in a 2nd or 3rd pass. One-line it shall be!